### DISCUSSION TITLE
Enabling hybrid RAG with JINA model bootloops docker image

### DISCUSSION THREAD

**Discussion by wbste** [open-webui/open-webui#7504] [Issues]
## Environment

- **Open WebUI Version:** [0.4.6] via Docker Compose.

## Description

**Bug Summary:**
Using the default SentenceTransformers on board. If I toggle Hybrid Search to **On**, then add jina reranker (tried both `jinaai/jina-reranker-v1-tiny-en` and `jinaai/jina-reranker-v1-turbo-en`), I get a bootloop condition where the docker image continuous tries to load the reranker (I believe). The recommended `BAAI/bge-reranker-v2-m3` performs fine, however.

The lower 3 lines are where it always crashes after the initial setting of the model. During each boot I do see `INFO  [open_webui.env] Reranking model set: jinaai/jina-reranker-v1-turbo-en`. To fix I have to open up `webui.db` and edit the `config` table's `data` field to change back the model to BAAI.

```
INFO  [open_webui.apps.retrieval.main] Updating reranking model: BAAI/bge-reranker-v2-m3 to jinaai/jina-reranker-v1-turbo-en
INFO  [open_webui.env] Saving 'RAG_RERANKING_MODEL' to the database

Fetching 19 files:   0%|          | 0/19 [00:00<?, ?it/s]
Fetching 19 files: 100%|██████████| 19/19 [00:00<00:00, 954.55it/s]
Container stopped
```

**Comment by tjbck** [comment]
Unable to reproduce here, keep us updated on your troubleshooting journey!

**Reply by wbste** [reply]
I'll give it another go. Thanks for taking the time!

**Comment by mahenning** [comment]
I have a similar problem, that using `jinaai/jina-reranker-v2-base-multilingual` as reranker leads to hanging of the app. Setting the reranker is no problem, but if I upload a pdf in the chat and query something, I get no output of the LLM and the UI gets unresponsive after some time and I have to restart the container. Using e.g .`BAAI/bge-reranker-v2-m3` as reranker works. I set `intfloat/multilingual-e5-base` as retrieve model.
The terminal logs of open webui shows nothing in that time frame, the browser logs (Firefox) shows
> Socket undefined disconnected due to ping timeout   
connect_error Error: timeout

Upon reloading the page I get
> TypeError: NetworkError when attempting to fetch resource.

Using e.g. `jinaai/jina-embeddings-v3` as embedding model with or without hybrid search works fine (No it doesn't, see edit). Using `jinaai/jina-reranker-v1-tiny-en` or `jinaai/jina-reranker-v1-turbo-en` also works for me.

Open-WebUI version: 0.4.8 in docker container with compose
Docker version: 27.1.1, build 6312585
Compose version: v2.29.1

Edit: I tried to re-set `jinaai/jina-embeddings-v3` as retrieval model, and when uploading a document in the chat I get the following message and the document is not used for the query:
> 'NoneType' object has no attribute 'encode'
